src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -182,11 +199,31 @@ public async Task<bool> TryRefreshAsync()
             {
                 await RefreshAsync().ConfigureAwait(false);
             }
-            catch (Exception e) when (
-                e is KeyVaultReferenceException ||
-                e is RequestFailedException ||
-                ((e as AggregateException)?.InnerExceptions?.All(e => e is RequestFailedException) ?? false) ||
-                e is OperationCanceledException)
+            catch (RequestFailedException e)
+            {
+                if (e.Status == (int)HttpStatusCode.Unauthorized || e.Status == (int)HttpStatusCode.Forbidden)
+                {
+                    _logger?.LogError(e, "Refresh operation failed due to an authentication error.");

```

Comment:
I am wondering if these should be logged as Warning. Single occurrence is not the end of day.

Reply:
But it's still an error that led to the failure of the current refresh operation. Warning might suggest that refresh succeeded with some transient errors.

Reply:
As discussed in the other comment, we consider all these failures transient. I'm putting myself in the shoes of the customer. If we use the config provider in our service, I will not want to see a level 4 error when this occurs just once. I may set up a monitor when this occurs repeatedly. So warning sounds more appropriate to me.